### PR TITLE
[New Rule] PowerShell Reflection Assembly Load

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Detects the use of Reflection.Assembly to load PEs and DLLs in memory in Powershell Scripts. Attackers use
this method to load executables and DLLs without writing to the disk, bypassing security solutions.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_posh_assembly_load.toml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+author = ["Elastic"]
+description = """
+This rule detects the use of Reflection.Assembly to load PEs and DLLs in memory in Powershell Scripts. Attackers use
+this method to load executables and DLLs without writing to the disk, bypassing security solutions.,
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-windows.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "PowerShell Reflection Assembly Load"
```
Comment (Samirbous) [inline rules/windows/defense_evasion_posh_assembly_load.toml:16]:
```suggestion
name = "Suspicious .NET Reflection via PowerShell"
```

File: rules/windows/defense_evasion_posh_assembly_load.toml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+]
+risk_score = 73
+rule_id = "e26f042e-c590-4e82-8e05-41e81bd822ad"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.code:"4104" and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_assembly_load.toml:28]: @w0rk3r  For this (and all rules using PS logging), can we convert from `event.code:4104` to `event.category:process`. We would even use the `event.action` in addition to the category if needed.

Comment (w0rk3r) [reply rules/windows/defense_evasion_posh_assembly_load.toml]: Done

File: rules/windows/defense_evasion_posh_assembly_load.toml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+[metadata]
+creation_date = "2021/10/15"
+maturity = "production"
+updated_date = "2021/10/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the use of Reflection.Assembly to load PEs and DLLs in memory in Powershell Scripts. Attackers use
+this method to load executables and DLLs without writing to the disk, bypassing security solutions.,
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_assembly_load.toml:10]:
```suggestion
this method to load executables and DLLs without writing to the disk, bypassing security solutions.
```

**Review by brokensound77 — APPROVED** [review]
One correction then LGTM 👍

**Pull request merged**